src/core/state.rs

```diff
@@ -74,64 +71,49 @@ pub fn config_read_v2(storage: &dyn Storage) -> ReadonlySingleton<StateV2> {
     singleton_read(storage, STATE_V2_KEY)
 }
 
-/// Boilerplate implementation of indexes for an IndexMap around state.
-/// This establishes a unique index on the scope spec address to ensure
-/// that saves cannot include duplicate scope specs.
-/// If it becomes a requirement in the future that we have duplicate scope specs,
-/// we will need to swap to a MultiIndex, and a lot of the lookups in the contract
-/// will fall apart.
-pub struct AssetDefinitionIndexesV2<'a> {
-    scope_spec: UniqueIndex<'a, String, AssetDefinitionV2>,
-}
-impl<'a> IndexList<AssetDefinitionV2> for AssetDefinitionIndexesV2<'a> {
-    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<AssetDefinitionV2>> + '_> {
-        let v: Vec<&dyn Index<AssetDefinitionV2>> = vec![&self.scope_spec];
-        Box::new(v.into_iter())
-    }
-}
-
+const ASSET_DEFINITIONS_V2_NAMESPACE: &str = "asset_definitions_v2";
```

Comment:
Might be worthwhile to just rename this to `ASSET_DEFINITIONS_NAMESPACE`.  Seems silly to create `ASSET_DEFINITIONS_V3` with the v2-named namespace.  I totally understand why you did it this way, so just a little name adjustment will likely do the trick.  Maybe also a comment as to why the V3 thing uses the v2 namespace?

Reply:
oh yeah, totally makes sense, changed